### PR TITLE
AWS Bedrock: add ability to inject BedrockRuntimeClient and BedrockRuntimeAsyncClient

### LINKED ISSUE #2238: [FEATURE] Add httpClient to bedrock runtime client builder
I need to use AWS Bedrock runtime client with a proxy setted on the httpClient, it is not possible right now because the BedrockRuntimeClient creation is in a private method, as a workaround and I am extending it on my company's project to add this httpClient. 

**Describe the solution you'd like**
A greater thing can be a possibility to pass the BedrockRuntimeClient itself to AbstractBedrockChatModel, without this, every time that I need to set a new property that is not on the AbstractSharedBedrockChatModel fields, I will need a new PR on langchain4j, by doing this if on some new version AWS put a new property it will automatically be ready for use on langchain4j, but I am not sure how can I pass a BedrockRuntimeClient.

Extending AbstractSharedBedrockChatModel to add a httpClient
image

So I can use it when creating a BedrockAnthropicMessageChatModel 
image

### LINKED ISSUE #2274: [BUG] Timeout on BedrockAnthropicChatMessageModel is not honored and time out prematurely.
**Describe the bug**
The timeout configuration on BedrockAnthropicChatMessageModel is not honored and time out prematurely.

**Log and Stack trace**
```
Retryable error detected. Will retry in 210ms. Request attempt number 4
2024-12-13 15:19:37 software.amazon.awssdk.core.exception.SdkClientException: Unable to execute HTTP request: Read timed out
2024-12-13 15:19:37     at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:111) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.exception.SdkClientException.create(SdkClientException.java:47) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.stages.utils.RetryableStageHelper.setLastException(RetryableStageHelper.java:223) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:83) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:36) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:56) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:36) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.executeWithTimer(ApiCallTimeoutTrackingStage.java:80) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:60) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:42) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallMetricCollectionStage.execute(ApiCallMetricCollectionStage.java:50) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallMetricCollectionStage.execute(ApiCallMetricCollectionStage.java:32) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:37) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:26) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.http.AmazonSyncHttpClient$RequestExecutionBuilderImpl.execute(AmazonSyncHttpClient.java:224) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.invoke(BaseSyncClientHandler.java:103) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.doExecute(BaseSyncClientHandler.java:173) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.lambda$execute$1(BaseSyncClientHandler.java:80) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.measureApiCallSuccess(BaseSyncClientHandler.java:182) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:74) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:45) ~[sdk-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:53) ~[aws-core-2.25.70.jar:?]
2024-12-13 15:19:37     at software.amazon.awssdk.services.bedrockruntime.DefaultBedrockRuntimeClient.invokeModel(DefaultBedrockRuntimeClient.java:224) ~[bedrockruntime-2.25.70.jar:?]
2024-12-13 15:19:37     at com.birst.genai.service.llm.chatmodel.bedrock.AbstractBirstBedrockChatModel.invoke(AbstractBirstBedrockChatModel.java:129) 
```

**To Reproduce**
Make the LLM take longer than 30 seconds to generate the output.
```
ChatLanguageModel model = BedrockAnthropicChatMessageModel.builder()
            .model("anthropic.claude-3-5-sonnet-20240620-v1:0")
            .timeout(Duration.ofMinutes(5))
            .region(bedrockconfig.getRegion())
            .temperature(0.0)
            .maxTokens(3000)
            .maxRetries(1)
            .build();
model.chat("make it generate something that takes longer than 30 seconds");
```

**Expected behavior**
The request should not timeout until it reaches the timeout configured.

**Please complete the following information:**
- LangChain4j version: 0.35.0
- LLM(s) used: BedrockAnthropicChatMessageModel
- Java version: 11

**Cause**
The cause for this issue is that we are not configuring the http client for the Bedrock client correctly. In `AbstractBedrockChatModel.java`, we need to explicitly set the timeout on the http client.
```
private BedrockRuntimeClient initClient() {
    return BedrockRuntimeClient.builder()
      .region(region)
      .credentialsProvider(credentialsProvider)
      .httpClient(ApacheHttpClient.builder()
        .socketTimeout(timeout)
        .build())
      .overrideConfiguration(c-> c.apiCallTimeout(timeout))
      .build();
  }
```
If the http client parameter is not set, then DefaultBedrockRuntimeClient will be used instead, which has a 30 second timeout on the socket.

### PR DESCRIPTION
**Pull request by omarmahamid**
## Issue
Closes #2274
Closes #2238

## General checklist
- [x] There are no breaking changes
- [x] I have added unit and/or integration tests for my change
- [ ] The tests cover both positive and negative cases
- [x] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [x] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green
- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

### CODE REVIEW

File: langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractBedrockChatModel.java [hunk 1]
```diff
@@ -127,6 +129,13 @@ protected InvokeModelResponse invoke(final InvokeModelRequest invokeModelRequest
         return getClient().invokeModel(invokeModelRequest);
     }
 
+    public BedrockRuntimeClient getClient() {
+        if (client == null) {
+            client = initClient();
```
Comment (langchain4j) [inline langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractBedrockChatModel.java]: This is not a proper way to do lazy initialization, there might be problems in concurrent scenarios. Is lazy initialization required at all? If not, I would remove lazy init logic. If yes, I would re-implement this taking concurrency into consideration.

**Review by ghost — COMMENTED** [review]
@omarmahamid thanks a lot!

File: langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractBedrockChatModel.java (hunk 1 above)
Comment (omarmahamid) [reply langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractBedrockChatModel.java]: fixed

by the way, thast's why i hate lombok. not convinent for this things

**Comment by rndtavares** [comment]
this PR algo closes issue #2238

**Review by ghost — APPROVED** [review]
@omarmahamid thank you!

**Pull request merged**